src/components/assets/AssetManagerDialog/AssetManagerForm.jsx

```diff
@@ -0,0 +1,163 @@
+    },
+    handleRemoveManager: () => {
+    }
+    
+  }
+}
+
+const validate = (values) => {
+  if (values.get('alreadyHave')) {
+    return {}
+  }
+  
+  let platformNameErrors = new ErrorList()
+  platformNameErrors.add(validator.name(values.get('platformName'), true))
+  
+  let platformAddressErrors = new ErrorList()
+  platformAddressErrors.add(validator.address(values.get('platformAddress'), true))
+  
+  return {
+    platformName: platformNameErrors.getErrors(),
+    platformAddress: platformAddressErrors.getErrors(),
+  }
+}
+
+const onSubmit = (/*values, dispatch*/) => {
+  // dispatch(createPlatform(values))
+}
+
+@connect(mapStateToProps, mapDispatchToProps)
+@reduxForm({form: FORM_ASSET_MANAGER, validate, onSubmit})
+export default class AssetManagerForm extends React.Component {
+  static propTypes = {
+    handleSubmit: PropTypes.func,
+    onClose: PropTypes.func,
+    formValues: PropTypes.object,
+    handleAddManager: PropTypes.func,
+    handleRemoveManager: PropTypes.func
+  }
+  
+  constructor (props) {
+    super(props)
+    
+    this.state = {
+      selectedManagers: {}
+    }
+  }
+  
+  handleSelectManager (managerAddress) {
+    const selectedManagers = {...this.state.selectedManagers}
+    selectedManagers[managerAddress] = !selectedManagers[managerAddress]
+    this.setState({selectedManagers})
+  }
+  
+  renderManagersList () {
+    const {selectedManagers} = this.state
+    return <div styleName='managersList'>
+      <div
+        styleName='managersListFieldRow'>
+        <div styleName='managersListFieldAddress'>
+          <div styleName={classnames('managersListIcon', 'xs-hide')}>
+            <i className='material-icons'>account_circle</i>
+          </div>
+          <Field
+            component={TextField}
+            name='managerAddress'
+            fullWidth
+            hintText={<Translate value={prefix('managerAddress')} />} />
+        </div>
+        <div styleName='managersListAction'>
+          <FloatingActionButton
+            mini
+            styleName={classnames('addManagerButton', 'xs-show')}
+            onTouchTap={() => {
+            }}>
+            <i className='material-icons'>add</i>
+          </FloatingActionButton>
+          <FlatButton
+            onTouchTap={() => this.props.handleAddManager()}
+            styleName={classnames('addManagerButton', 'xs-hide')}
+            label={<Translate value={prefix('addManagersButton')} />}
+          />
+        </div>
+      </div>
+      {
+        managersList.map(
+          item => (
+            <div key={item} styleName={classnames('managersListRow', {'selected': selectedManagers[item]})}>
+              <div
+                onTouchTap={() => this.handleSelectManager(item)}
+                styleName='managersListAddress'>
+                <div styleName='managersListIcon'>
+                  <i className='material-icons'>account_circle</i>
+                </div>
+                <div styleName='ellipsis'>
```

Comment:
for parent add 'display: flex', for children 'flex-grow: 0 or 1', for ellipsised elements add ellipsis. This will wrap it.